src/platform/qt/MemoryView.cpp

```diff
@@ -258,9 +256,22 @@ void MemoryView::updateSelection(uint32_t start, uint32_t end) {
 void MemoryView::updateStatus() {
 	unsigned align = m_ui.hexfield->alignment();
 	mCore* core = m_controller->thread()->core;
-	QByteArray selection(m_ui.hexfield->serialize());
-	QString text(m_ui.hexfield->decodeText(selection));
-	m_ui.stringVal->setText(text);
+	QByteArray txt;
+	uint8_t chr;
+	int i = 0;
+	do {
+		chr = core->rawRead8(core, m_selection.first + i, m_ui.segments->value());
+		txt.append(chr);
+		++i;
+	} while (chr);
+
+	QString enc = m_ui.encoding->currentText();
+	if (enc == "ASCII") {
+		m_ui.stringVal->setText(QString::fromLatin1(txt.constData()));
+	} else {
+		QTextCodec* codec = QTextCodec::codecForName(enc.toLatin1());
+		m_ui.stringVal->setText(codec->toUnicode(txt.constData()));
+	}
```

Comment:
Absolutely not. This is handled in decodeText for a reason--it's the model, not the view. Not to mention, it's already handling text codecs for TBL files. You've just bypassed this logic instead of augmenting it.